atomic_reactor/build.py

```diff
@@ -146,7 +146,7 @@ def __init__(self, source, image, **kwargs):
         self.original_base_image = None
         self._base_image_inspect = None
         self._parents_pulled = False
-        self.parent_images = {}  # dockerfile image => locally available image
+        self.parent_images = {}  # dockerfile ImageName => locally available ImageName
```

Comment:
There are a few more places where this change has to be taken into account e.g.

https://github.com/projectatomic/atomic-reactor/blob/f4122f110b7b4d53dcf499cdecd381a21ab9b30b/atomic_reactor/plugins/exit_store_metadata_in_osv3.py#L236

ImageName is not JSON serializable, and we would want the string in the output anyway.

Reply:
So my first thought was to just make ImageName JSON serializable, but a couple hours of banging my head against Python inheritance and overriding made me rethink that.

I added a little function that converts a dict of parent images into a dict of parent image strings.

Reply:
https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/plugins/pre_koji_parent.py#L83-L84
pulls the image name out of `parent_images` and calls
https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/build.py#L223
which is expecting a string - think that needs to be changed.

Reply:
Fixed.